layouts/partials/lesson-gallery.html

```diff
@@ -0,0 +1,58 @@
+{{- $show := .show | default "topics" -}}
+
+<div class="lesson_gallery" data-show="{{ $show }}">
+  <p class="lesson_gallery_controls center">
+    {{
+      partial "clickable"
+      (dict
+        "text" "Featured"
+        "onclick" "lessonGalleryTab(this, 'featured');"
+        "active" (eq $show "featured")
+      )
+    }}
+    {{
+      partial "clickable"
+      (dict
+        "text" "By Topic"
+        "onclick" "lessonGalleryTab(this, 'topics');"
+        "active" (eq $show "topics")
+      )
+    }}
+    {{
+      partial "clickable"
+      (dict
+        "text" "By Date"
+        "onclick" "lessonGalleryTab(this, 'date');"
+        "active" (eq $show "date")
+      )
+    }}
+  </p>
+
+  {{ $pages := site.Pages }}
+  
+  <div class="lesson_gallery_tab" data-tab="featured">
+    {{ range site.Data.featured }}
+      {{ partial "lesson-card" (dict "id" .) }}
+    {{ end }}
+  </div>
+
+  <div class="lesson_gallery_tab" data-tab="topics">
+    {{ range site.Data.topics }}
+      {{ partial "topic-card" (dict "name" .name) }}
+      <div class="lessons">
+        {{ range .lessons }}
+          {{ partial "lesson-card" (dict "id" .) }}
```

Comment:
If I understand correctly, here's how this would change things. We'd need:

```
▾ layouts/
  ▾ lessons/
      card.html
      card-mini.html
      card-mini-prev.html
      card-mini-prev.html
      tooltip.html
      single.html
      ...more?
```

Several duplicates, though each file would only be a line or two.

Then this:
```
{{ range site.Data.featured }}
  {{ partial "lesson-card" (dict "id" .) }}
{{ end }}
```

would become something like this?
```
{{ range site.Data.featured }}
  {{ partial "lesson-card" (site.GetPage .) }}
{{ end }}
```

I think this just adds files and doesn't really make anything cleaner.

The scratch pad idea may be workable. But I think it works fine as it is now. lesson-card is a component. If this were react, it'd just be a component that you render with different options in different places.

Reply:
Definitely fine for now. It would actually look something like:

```
{{ range site.Data.featured }}
  {{ (site.GetPage .).Render "card" }}
{{ end }}
```

I agree that it adds extra files. Although, one long term benefit worth noting is that when you go to update the "lesson-card" partial in some way or another, rather than doing a "find/replace" for all of the places the partial is used, you can just update the content view once (even dramatically) and feel confident that the changes are propagated everywhere the lesson card content view is rendered.

Again, more than fine for now.